packages/trpc/server/routers/viewer/organizations/adminDelete.handler.ts

```diff
@@ -0,0 +1,70 @@
+import { deleteDomain } from "@calcom/lib/domainManager/organization";
+import { prisma } from "@calcom/prisma";
+
+import { TRPCError } from "@trpc/server";
+
+import type { TrpcSessionUser } from "../../../trpc";
+import type { TAdminDeleteInput } from "./adminDelete.schema";
+
+type AdminDeleteOption = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+  input: TAdminDeleteInput;
+};
+
+export const adminDeleteHandler = async ({ input }: AdminDeleteOption) => {
+  const foundOrg = await prisma.team.findUnique({
+    where: {
+      id: input.orgId,
+      metadata: {
+        path: ["isOrganization"],
+        equals: true,
+      },
+    },
+    include: {
+      members: {
+        select: {
+          user: true,
+        },
+      },
+    },
+  });
+
+  if (!foundOrg)
+    throw new TRPCError({
+      code: "FORBIDDEN",
+      message: "Organization not found",
+    });
+
+  if (foundOrg.slug) {
+    await deleteDomain(foundOrg.slug);
+  }
+
+  await renameUsersToAvoidUsernameConflicts(foundOrg.members.map((member) => member.user));
+  await prisma.team.delete({
+    where: {
+      id: input.orgId,
+    },
+  });
+
+  return {
+    ok: true,
+    message: `Organization ${foundOrg.name} deleted.`,
+  };
+};
+
+export default adminDeleteHandler;
+
+async function renameUsersToAvoidUsernameConflicts(users: { id: number; username: string | null }[]) {
+  for (const user of users) {
+    await prisma.user.update({
+      where: {
+        id: user.id,
+      },
+      data: {
+        username: `${user.username || ""}-${user.id}`,
```

Comment:
I think we should actually specific this -> You can very easily get a cal account with no username set and having a user with `-519` isnt something we normally allow i dont think 

```suggestion
        username: `${user.username || "nameless_user"}-${user.id}`,
```

Reply:
What's the case when there is no username ? Does it happen when we invite a user?

I actually wanted to avoid `nameless` but I think it is a better approach when we know for sure that having null username is possible.

Reply:
Its very rare but you can replicate it by
1) invite a new user
2) new user doesnt want to join a team 
3) resets password
4) logs in 

We should probably work on re-directing to onboarding if no username is set. Or ensure user.username exists bfore we delete them? Not sure the best approach around that